kafkatjenester/dokumenthendelse/src/main/java/no/nav/foreldrepenger/melding/kafkatjenester/dokumenthendelse/DokumentMeldingConsumer.java

```diff
@@ -0,0 +1,117 @@
+package no.nav.foreldrepenger.melding.kafkatjenester.dokumenthendelse;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Properties;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.serialization.StringDeserializer;
+
+import no.nav.vedtak.konfig.KonfigVerdi;
+
+@ApplicationScoped
+public class DokumentMeldingConsumer {
+
+    private static final int TIMEOUT = 10000;
+    KafkaConsumer<String, String> kafkaConsumer;
+    String topic;
+
+    public DokumentMeldingConsumer() {
+        //CDI
+    }
+
+    @Inject
+    public DokumentMeldingConsumer(@KonfigVerdi("kafka.dokumenthendelse.topic") String topic,
+                                   @KonfigVerdi("bootstrap.servers") String bootstrapServers,
+                                   @KonfigVerdi("kafka.dokumenthendelse.schema.registry.url") String schemaRegistryUrl,
+                                   @KonfigVerdi("kafka.dokumenthendelse.group.id") String groupId,
+                                   @KonfigVerdi("systembruker.username") String username,
+                                   @KonfigVerdi("systembruker.password") String password) {
+
+        Properties properties = new Properties();
+        properties.put("bootstrap.servers", bootstrapServers);
+        properties.put("schema.registry.url", schemaRegistryUrl);
+        properties.put("group.id", groupId);
+        properties.put("enable.auto.commit", "false");
+        properties.put("max.poll.records", "1");
+
+        setSecurity(username, properties);
+
+        addUserToProperties(username, password, properties);
+
+        this.kafkaConsumer =
+
+                createConsumer(properties);
+        this.topic = topic;
+
+        subscribe();
+
+    }
+
+    void setSecurity(String username, Properties properties) {
+        if (username != null && !username.isEmpty()) {
+            properties.put("security.protocol", "SASL_SSL");
+            properties.put("sasl.mechanism", "PLAIN");
+        }
+    }
+
+
+    void addUserToProperties(@KonfigVerdi("kafka.username") String username, @KonfigVerdi("kafka.password") String password, Properties properties) {
+        if (username != null && !username.isEmpty() && password != null && !password.isEmpty()) {
```

Comment:
Kunne kanskje hatt en "nonEmpty" helper metode for alle disse? Ellers ingen innvendinger.

Reply:
Det finnes vel forsåvidt, denne koden er bare kopiert fra felles